### ISSUE TITLE
Crashes with ExternalName service

### ISSUE THREAD

**Issue by abh** [jcmoraisjr/haproxy-ingress#336]
In an older cluster I had a service with an externalName service. Moving to a new cluster with a newer haproxy-ingress, the ingress crashes with this.

```
{
    "kind": "Service",
    "apiVersion": "v1",
    "metadata": {
        "name": "github"
    },
    "spec": {
        "type": "ExternalName",
        "externalName": "www.github.com",
        "ports": [{ "port": 80 }]
    }
}```

**Comment by abh** [comment]
This makes it not crash (but doesn't make the ExternalService work).

```
diff --git a/pkg/controller/dynconfig/dynconfig.go b/pkg/controller/dynconfig/dynconfig.go
index e38ad8d..508b9c1 100644
--- a/pkg/controller/dynconfig/dynconfig.go
+++ b/pkg/controller/dynconfig/dynconfig.go
@@ -18,6 +18,7 @@ package dynconfig
 
 import (
 	"fmt"
+	"log"
 	"reflect"
 	"sort"
 
@@ -359,6 +360,10 @@ func (d *DynConfig) fillBackendServerSlots() {
 			for _, endpoint := range d.updBackendsMap[backendName].Endpoints {
 				curEndpoint := endpoint
 				target := fmt.Sprintf("%s:%s", endpoint.Address, endpoint.Port)
+				if endpoint.Target == nil {
+					log.Printf("nil endpoint for %s %q", backendName, target)
+					continue
+				}
 				backendServerName := endpoint.Target.Name
 				if len(backendServerName) == 0 {
 					backendServerName = target
```

**Comment by abh** [comment]
This seems to work with the 0.10 beta now. :-)

**Comment by jcmoraisjr** [comment]
Hi, I definitely missed this issue; need to make a triage on all of the opened ones. Sorry about taking so long.

This was fixed on #455 and should work fine on the latest tag of all branches - currently v0.8.4, v0.9, and also v0.10-beta.1 as you pointed out.

Thanks for the report and feedback.
